server/backends/blobstore/util/util.go

```diff
@@ -67,6 +69,30 @@ var BlobPath = func(blobName string) string {
 	return filepath.Join(*pathPrefix, blobName)
 }
 
+type prefixBlobstore struct {
+	interfaces.Blobstore
+}
+
+func NewPrefixBlobstore(b interfaces.Blobstore) *prefixBlobstore {
```

Comment:
nit: it might be worth having a method `WithPrefix(b)` that just returns `b` if prefix is empty, else returns `NewPrefixBlobstore(b)`. That way we avoid the `prefixBlobstore` overhead for the common case of not setting a prefix.